codespell_lib/_codespell.py

```diff
@@ -984,6 +985,18 @@ def parse_file(
     return bad_count
 
 
+def flatten_comma_separated_arguments(
+    arguments: List[str],
+) -> List[str]:
+    """
+    >>> flatten_comma_separated_arguments(["a,b,c", "d", "e"])
+    ['a', 'b', 'c', 'd', 'e']
+    >>> flatten_comma_separated_arguments([])
+    []
+    """
+    return [item for argument in arguments for item in argument.split(",")]
```

Comment:
```suggestion
    >>> flatten_comma_separated_arguments(None)
    []
    """
    return [item for argument in arguments or [] for item in argument.split(",")]
```

Reply:
I had added `or []` in the arguments of `flatten_comma_separated_arguments`, but it's much better this way., it is better to make the function itself lenient.

Reply:
But then it means we can pass `None` to a function that is documented to accept `List(str)` only. Again, I am puzzled by these type hints.

Reply:
You are not alone.  The tools are not perfect but the situation is better than before type hints.  The more hints you put into your functions and their return types, the better the type checkers get and finding issues.  Who is [the fifth most prolific contributor to `mypy`](https://github.com/python/mypy/graphs/contributors)?